BEIMA.Backend.Test/MongoService/DeviceTest.cs

```diff
@@ -265,11 +193,11 @@ public void DeviceNotInstantiated_InstantiateUsingFullConstructorWithAllNullStri
         }
 
         [Test]
-        public void DeviceInstantiatedWithSomeNullValuesWithFullConstructor_CallGetBsonDocument_ExceptionIsThrown()
+        public void DeviceInstantiatedWithSomeNullValuesWithFullConstructor_CallGetBsonDocument_NoExceptionIsThrown()
         {
             var device = new Device(validObjId, validDeviceTypeId, validDeviceTag, validManufacturer, null, null, validYearManufactured, null);
             device.SetLastModified(validDate, validUser);
-            Assert.Throws<ArgumentNullException>(() => { device.GetBsonDocument(); });
+            Assert.DoesNotThrow(() => { device.GetBsonDocument(); });
```

Comment:
This needs to be revised back. The reason why this should throw an exception is because SetLocation was not called yet.

Reply:
Added back